services/staging/rs_server_staging/main.py

```diff
@@ -56,6 +58,101 @@
 api = API(get_config(os.environ["PYGEOAPI_CONFIG"]), os.environ["PYGEOAPI_OPENAPI"])
 
 
+def init_db():
+    """Initialize the PostgreSQL database connection and sets up required table and ENUM type.
+
+    This function constructs the database URL using environment variables for PostgreSQL
+    credentials, host, port, and database name. It then creates an SQLAlchemy engine and
+    registers the ENUM type EStagingStatus and the 'job' tables if they don't already exist.
+
+    Environment Variables:
+        - POSTGRES_USER: Username for database authentication.
+        - POSTGRES_PASSWORD: Password for the database.
+        - POSTGRES_HOST: Hostname of the PostgreSQL server.
+        - POSTGRES_PORT: Port number of the PostgreSQL server.
+        - POSTGRES_DB: Database name.
+
+    Raises:
+        RuntimeError: If any of the required environment variables are missing or if an SQLAlchemy
+                      error occurs while creating tables or ENUM types.
+
+    Returns:
+        None
+
+    Exceptions:
+        - **KeyError**: Raised when any of the required environment variables is missing.
+        - **SQLAlchemyError**: Raised for any database errors encountered while creating tables
+          or ENUM types.
+    """
+    try:
+        # pylint: disable=consider-using-f-string
+        database_url = "postgresql+psycopg2://{user}:{password}@{host}:{port}/{dbname}".format(
```

Comment:
@agrosu1978 the same code is implemented in https://github.com/RS-PYTHON/rs-server/blob/develop/services/common/rs_server_common/db/database.py, did you try to reuse this module ? 

Also this module checks (with a file lock) that multiple services do not try to init the db at the same time. In rs-demo are you using the same postgres instance for staging, adgs and cadip ? If yes, maybe we'll run into the same problem again randomly if you do not reuse rs_server_common/db/database.py (even if I know that adgs/cadip database will be removed soon). 

And rs_server_common/db/database.py also calls `Base.metadata.drop_all(bind=self._engine)`, I don't know if it's useful to call it from the staging.